src/main/cpp/states/Music.cpp

```diff
@@ -0,0 +1,107 @@
+#include <states/Music.h>
+
+int Music::GetButton() 
+{
+    for (int i = 1; i < 9; i++) 
+    {
+        if (m_joystick->GetRawButton(i))

```

Comment:
don't reference joystick directly.  In fact, do tie it to a joystick at all.   Have an interface where maybe each song is assigned an enum value and have callers call a method like Play( enum MUSIC_TYPE).